scripts/ecs-anywhere-install.sh

```diff
@@ -0,0 +1,526 @@
+            fail
+            ;;
+        esac
+
+        case "$PKG_MANAGER" in
+        dnf)
+            dnf install -y docker
+            ;;
+        yum)
+            yum install -y docker
+            ;;
+        apt)
+            apt install -y docker.io
+            ;;
+        zypper)
+            zypper install -y docker
+            ;;
+        esac
+        ;;
+    none)
+        echo "Docker install source is none, not installing docker"
+        ;;
+    esac
+
+    ok
+}
+
+install-ecs-agent() {
+    try "install ecs agent"
+    if [ -x "/usr/libexec/amazon-ecs-init" ]; then
+        echo "ecs agent is already installed"
+        # TODO upgrade the agent?
+        ok
+        return
+    fi
+
+    local dir
+    dir="$(mktemp -d)"
+    case "$PKG_MANAGER" in
+    apt)
+        curl -o "$dir/$DEB_PKG_NAME" "$DEB_URL"
+        if $CHECK_SHA; then
+            curl -o "$dir/$DEB_PKG_NAME.asc" "$DEB_URL.asc"
+            ecs-init-signature-verify "$dir/$DEB_PKG_NAME.asc" "$dir/$DEB_PKG_NAME"
+        fi
+        apt install -y "$dir/$DEB_PKG_NAME"
+        rm -rf "$dir"
+        ;;
+    dnf | yum | zypper)
+        curl -o "$dir/$RPM_PKG_NAME" "$RPM_URL"
+        if $CHECK_SHA; then
+            curl -o "$dir/$RPM_PKG_NAME.asc" "$RPM_URL.asc"
+            ecs-init-signature-verify "$dir/$RPM_PKG_NAME.asc" "$dir/$RPM_PKG_NAME"
+        fi
+        local args="-y"
+        if [[ "$PKG_MANAGER" == "zypper" ]]; then
+            args="${args} --allow-unsigned-rpm"
+        fi
+        $PKG_MANAGER install ${args} "$dir/$RPM_PKG_NAME"
+        rm -rf "$dir"
+        ;;
+    esac
+
+    if [ ! -f "/etc/ecs/ecs.config" ]; then
+        touch /etc/ecs/ecs.config
+    else
+        echo "/etc/ecs/ecs.config already exists, preserving existing config and appending cluster name."
+    fi
+    echo "ECS_CLUSTER=$ECS_CLUSTER" >>/etc/ecs/ecs.config
+
+    if [ ! -f "/var/lib/ecs/ecs.config" ]; then
+        touch /var/lib/ecs/ecs.config
+    else
+        echo "/var/lib/ecs/ecs.config already exists, preserving existing config and appending ECS anywhere requirements."
+    fi
+    echo "AWS_DEFAULT_REGION=$REGION" >>/var/lib/ecs/ecs.config
+    echo "ECS_EXTERNAL=true" >>/var/lib/ecs/ecs.config
+    if [ ! -z "$ECS_ENDPOINT" ]; then
+        echo "ECS_BACKEND_HOST=$ECS_ENDPOINT" >>/var/lib/ecs/ecs.config
+    fi
+    systemctl enable ecs
+    systemctl start ecs
+
+    ok
+}
+
+ecs-init-signature-verify() {
+    echo "Verifying the signature of ECS Init Artifact"
+
+    #TODO [Update before release] Update links here to use prod urls (or urls specified by $DEB_URL or $RPM_URL)
+    curl -o "$dir/amazon-ecs-init.gpg" "https://ecs-init-packages-testing.s3.amazonaws.com/amazon-ecs-public-key.gpg"
+    gpg --import "$dir/amazon-ecs-init.gpg"
+
+    gpg --verify "$1" "$2" &> "$dir/verifyOut"
+
+    if grep -Fxq "BAD signature" "$dir/verifyOut"; then
```

Comment:
same comment as I made for the ssm signature verification (https://github.com/aws/amazon-ecs-init/pull/398/files#r596305112), we should check for the gpg command to error directly rather than grepping for errors.

Reply:
Updated.